### PR TITLE
Add new support bdcom

### PR DESCRIPTION
**Pull request by hrtrd**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/definitions/discovery/bdcom.yaml [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
-                num_oid: '.1.3.6.1.4.1.3320.9.109.1.1.1.1.5.{{ $index }}'
-                index: 0
+                oid: nmscardCPUUtilization
+                num_oid: '.1.3.6.1.4.1.3320.3.6.10.1.11.{{ $index }}'
+    sensors:
+        temperature:
+            data:
+                -
+                    oid: nmscardTemperature
+                    num_oid: '.1.3.6.1.4.1.3320.3.6.10.1.13.{{ $index }}'
```
Comment (hrtrd) [inline includes/definitions/discovery/bdcom.yaml:23]: Mempool, os and CPU works fine.  

Sensor temp, and voltage problem.  I have attached a mib file, the command is given a separate answer.

Expected "(" (_): At line 287 in /opt/librenms/mibs/bdcom/NMS-CHASSIS
Should be ACCESS (1FastEthernet-100M): At line 287 in /opt/librenms/mibs/bdcom/NMS-CHASSIS
Bad parse of OBJECT-TYPE: At line 287 in /opt/librenms/mibs/bdcom/NMS-CHASSIS
nmscardCPUUtilization: Unknown Object Identifier (Sub-id not found: (top) -> nmscardCPUUtilization)  
sensors SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'COMMUNITY' '-OteQUs' '-Ih' '-m' 'NMS-CHASSIS' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/bdcom' 'udp:HOSTNAME:161' 'nmscardTemperature']
Exitcode: 1  
Expected "(" (_): At line 287 in /opt/librenms/mibs/bdcom/NMS-CHASSIS
Should be ACCESS (1FastEthernet-100M): At line 287 in /opt/librenms/mibs/bdcom/NMS-CHASSIS
Bad parse of OBJECT-TYPE: At line 287 in /opt/librenms/mibs/bdcom/NMS-CHASSIS
nmscardTemperature: Unknown Object Identifier (Sub-id not found: (top) -> nmscardTemperature)  
  
Expected "(" (_): At line 287 in /opt/librenms/mibs/bdcom/NMS-CHASSIS
Should be ACCESS (1FastEthernet-100M): At line 287 in /opt/librenms/mibs/bdcom/NMS-CHASSIS
Bad parse of OBJECT-TYPE: At line 287 in /opt/librenms/mibs/bdcom/NMS-CHASSIS
nmscardTemperature: Unknown Object Identifier (Sub-id not found: (top) -> nmscardTemperature)

Comment (murrant) [reply includes/definitions/discovery/bdcom.yaml:23]: net-snmp can't parse the mib.  You'll need to alter the MIB to make it work.

Comment (hrtrd) [reply includes/definitions/discovery/bdcom.yaml:23]: "murrant30.11.2020
@hartred the mib is invalid. If you want it to work, you need to replace the mentioned _ with -"    can u show some example?

Comment (hrtrd) [reply includes/definitions/discovery/bdcom.yaml:23]: Fix that. Edit yaml now and re-upload later

**Comment by hrtrd** [comment]
@Jellyfrog @murrant can u MR pls? )

**Comment by Jellyfrog** [comment]
@hrtrd please check failing tests; https://github.com/librenms/librenms/pull/12370/checks?check_run_id=1640259808

**Comment by hrtrd** [comment]
> @hrtrd please check failing tests; https://github.com/librenms/librenms/pull/12370/checks?check_run_id=1640259808

oh, new tests? 

that need new json ?  "OS bdcom: Discovered processors data does not match that found in tests/data/bdcom.json
Failed asserting that two arrays are equal."

**Comment by hrtrd** [comment]
+            'processor_oid' => '.1.3.6.1.2.1.25.3.3.1.2.1'     where does this oid come from? it is not in json or discovery

**Comment by murrant** [comment]
@hrtrd that is from the HOST-RESOURCES-MIB, you sure your dev environment is up to date?

**Comment by hrtrd** [comment]
> @hrtrd that is from the HOST-RESOURCES-MIB, you sure your dev environment is up to date?

try it rn. 
./scripts/composer_wrapper.php install  then pip3 install requirements

**Comment by murrant** [comment]
By dev environment up to date I mean.

```
git fetch upstream
git rebase upstream/master or git merge upstream/master
```

**Comment by hrtrd** [comment]
> By dev environment up to date I mean.
> 
> ```
> git fetch upstream
> git rebase upstream/master or git merge upstream/master
> ```

yup, all updated and 
./lnms dev:check unit --db --snmpsim -o bdcom
Only checking os: bdcom
Running unit check... 
Starting snmpsim listening on 127.1.6.2:1162... 
Refreshing database...done
PHPUnit 9.5.0 by Sebastian Bergmann and contributors.

........                                                            8 / 8 (100%)

Time: 00:45.419, Memory: 80.50 MB

OK (8 tests, 1659 assertions)

**Pull request closed without merging**